Automattic/wp-calypso
# ci(e2e): Clean up config

#### Changes proposed in this Pull Request

General cleanup:

* Rename `magellan-gutenberg.json` to `magellan-wpcom.json`
* Clean up redundant magellan CLI args.
* Move Calypso e2e tests under `specs/specs-calypso`
* Move Gutenberg/WPCOM e2e tests under `specs/specs-wpcom`
* Move Playwrith tests under `specs/specs-playwright` (cc @worldomonation)

See individual comments for more details.

#### Testing instructions

* Verify e2e tests pass
* Verify Gutenberg e2e tests pass and it detects the same number of tests than `trunk`
* Verify Playwright tests pass and it detects the same number of tests than `trunk`
